### PR TITLE
Replace `DataId` with `Link`

### PR DESCRIPTION
**Pull request by PatrykWalach**
The demos are working, the question is, should I replace all the variable names as well `recordId` -> `recordLink`, `rootId` -> `rootLink`, `targetParentRecordId` -> `targetParentRecordLink`

### CODE REVIEW

**Comment by PatrykWalach** [comment]
I've renamed variables in the second commit, if we don't want this then we can revert, but to me it makes sense to rename them

File: demos/github-demo/src/isograph-components/__isograph/User/RepositoryConnection/refetch_reader.ts [hunk 1]
```diff
@@ -1,17 +1,17 @@
-import type { RefetchReaderArtifact, ReaderAst, RefetchQueryNormalizationArtifact } from '@isograph/react';
+import type { ReaderAst, RefetchQueryNormalizationArtifact, RefetchReaderArtifact } from '@isograph/react';
```
Comment (rbalicki2) [inline demos/github-demo/src/isograph-components/__isograph/User/RepositoryConnection/refetch_reader.ts:1]: looks like `pnpm build-demos` needs to be run to fix this

File: demos/pet-demo/src/components/__isograph/AdItem/AdItemDisplay/refetch_reader.ts [hunk 2]
```diff
@@ -1,17 +1,17 @@
-import type { RefetchReaderArtifact, ReaderAst, RefetchQueryNormalizationArtifact } from '@isograph/react';
+import type { ReaderAst, RefetchQueryNormalizationArtifact, RefetchReaderArtifact } from '@isograph/react';
 const includeReadOutData = (variables: any, readOutData: any) => {
   variables.id = readOutData.id;
   return variables;
 };
 
-import { makeNetworkRequest, wrapResolvedValue, type IsographEnvironment, type FragmentReference, type RefetchQueryNormalizationArtifactWrapper, type DataId, type TopLevelReaderArtifact } from '@isograph/react';
+import { makeNetworkRequest, wrapResolvedValue, type FragmentReference, type IsographEnvironment, type Link, type RefetchQueryNormalizationArtifactWrapper, type TopLevelReaderArtifact } from '@isograph/react';
```
Comment (rbalicki2) [inline demos/pet-demo/src/components/__isograph/AdItem/AdItemDisplay/refetch_reader.ts:7]: why is this changing? this comes from a call to `format!`, but we don't re-order `type IsographEnvironment` and `type FragmentReference` anywhere??

Comment (PatrykWalach) [reply demos/pet-demo/src/components/__isograph/AdItem/AdItemDisplay/refetch_reader.ts]: This got changed by my editor, it's fixed

**Review by rbalicki2 — APPROVED** [review]
lgtm, just one variable name capitalization nit and we're good to land

**Review by rbalicki2 — APPROVED** [review]
LGTM shipittt

**Pull request merged**